src/test/java/bio/terra/cloudfiletodatastore/testrunner/cloudfunctions/MockTestRunnerStreamingProcessor.java

```diff
@@ -0,0 +1,84 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveException;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.compressors.CompressorException;
+
+public class MockTestRunnerStreamingProcessor extends MessageProcessor {
+  private static final Logger logger =
+      Logger.getLogger(MockTestRunnerStreamingProcessor.class.getName());
+
+  public MockTestRunnerStreamingProcessor(FileUploadedMessage fileUploadedMessage) {
+    super(fileUploadedMessage);
+  }
+
+  @Override
+  public void processMessage() {
+    String sourceBucket = message.getSourceBucket();
+    String resourceName = message.getResourceName();
+    String projectId = CF_ENV.get("GCLOUD_PROJECT");
+    String dataSet = CF_ENV.get("BQ_DATASET");
+    String table = CF_ENV.get("BQ_TABLE");
+
+    InputStream in = MOCK_TGZ;
+
+    // The intermediate step is necessary for ArchiveInputStream to recognize that the file being
+    // processed is a tar.gz
+    try (ArchiveInputStream archiveInputStream =
+        MediaTypeUtils.createArchiveInputStream(MediaTypeUtils.createCompressorInputStream(in))) {
+      /*
+       * ArchiveInputStream is a special type of InputStream that emits an EOF when it gets to the end
+       * of a file in the archive. Once it’s done, call getNextEntry to reset the stream and start
+       * reading the next file. When getNextEntry returns null, you’re at the end of the archive.
+       */
+      ArchiveEntry archiveEntry;
+      while ((archiveEntry = archiveInputStream.getNextEntry()) != null) {
+        if (archiveEntry.isDirectory()) {
+          continue;
+        }
+
+        if (!Files.getNameWithoutExtension(archiveEntry.getName()).equals(table)) {
+          logger.log(
+              Level.WARNING,
+              String.format(
+                  "Processing skipped for project %s: bucket %s and resource name %s.",
+                  projectId, sourceBucket, resourceName));
+          continue;
+        }
+        logger.log(
+            Level.INFO,
+            String.format(
+                "Processing %s (%s bytes) for streaming to BQ table %s",
+                archiveEntry.getName(),
+                archiveEntry.getSize(),
+                Files.getNameWithoutExtension(archiveEntry.getName())));
+
+        byte[] data = archiveInputStream.readAllBytes();
+        streamToBQ(projectId, dataSet, table, data);
+      }
+    } catch (ArchiveException | IOException | CompressorException e) {
+      logger.log(Level.SEVERE, e.getMessage());
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void streamToBQ(String projectId, String dataset, String table, byte[] data) {
+    assertEquals("terra-kernel-k8s", projectId);
+    assertEquals("simple_data_set", dataset);
+    assertEquals("SUMMARY_testRun", table);
+    assertEquals(1350, data.length);
+  }
+}
```

Comment:
I am not going to hold this PR any longer for this comment, but for a follow-on refactoring: `MockTestRunnerStreamingProcessor` and `MockTestRunnerStreamerFunction` are so close to `TestRunnerStreamingProcessor` and `TestRunnerStreamingFunction`, and there is so much copy-pasted between them, I'd like to find a better way to handle them so that the tests don't get out of sync with the runtime implementation.

For instance, if `TestRunnerStreamingProcessor` had a method:
```
  void streamToBQ(String projectId, String dataSet, String table, byte[] data) throws IOException {
    BigQueryUtils.streamToBQ(projectId, dataSet, table, data);
  }
```
instead of inlining the `BigQueryUtils.streamToBQ(…)` call inside `processMessage()`, then `MockTestRunnerStreamingProcessor` could just extend `TestRunnerStreamingProcessor` and override the `streamToBQ` instead of copy-pasting the entire `processMessage` implementation.